src/Be.Vlaanderen.Basisregisters.AcmIdm/AddAuthenticationExtensions.cs

```diff
@@ -27,7 +30,17 @@ public static class AddAuthenticationExtensions
             this IServiceCollection services,
             OAuth2IntrospectionOptions oAuth2IntrospectionOptions)
         {
+            var nisCodeServiceUrl = services
+                .BuildServiceProvider()
+                .GetRequiredService<IConfiguration>()
+                .GetValue<string>("NisCodeServiceUrl");
+            if (string.IsNullOrWhiteSpace(nisCodeServiceUrl))
+            {
+                throw new ConfigurationErrorsException("Configuration should have a value for \"NisCodeServiceUrl\".");
+            }
+
             return services
+                .AddHttpProxyNisCodeService(nisCodeServiceUrl)
```

Comment:
Shouldn't we register it when registering authorization (instead of authentication)? So here
https://github.com/Informatievlaanderen/basisregisters-acmidm/blob/main/src/Be.Vlaanderen.Basisregisters.AcmIdm/AddAuthorizationExtensions-MinimalApi.cs#L9
and
here https://github.com/Informatievlaanderen/basisregisters-acmidm/blob/main/src/Be.Vlaanderen.Basisregisters.AcmIdm/AddAuthorizationExtensions-WebApi.cs#L23

Reply:
authentication is always needed so i chose to do it at authentication